finality-aleph/src/network/io.rs

```diff
@@ -1,33 +1,42 @@
 use futures::channel::mpsc;
 
 use crate::network::{
-    manager::NetworkData, ConnectionManagerIO, Data, Multiaddress, NetworkServiceIO as NetworkIo,
-    SessionManagerIO,
+    manager::{DataInSession, VersionedAuthentication},
+    ConnectionManagerIO, Data, Multiaddress, NetworkServiceIO as NetworkIO, SessionManagerIO,
 };
 
-type NetworkServiceIO<D, M> = NetworkIo<NetworkData<D, M>, M>;
+type AuthenticationNetworkIO<D, M> = NetworkIO<VersionedAuthentication<M>, DataInSession<D>, M>;
 
 pub fn setup<D: Data, M: Multiaddress + 'static>() -> (
```

Comment:
I think a good thing to add somewhere, maybe here, would be a comment about what is `network`, `io`, `user`, `service`. It could explain a lot for someone reading this code. Maybe not necessarily in this PR, but should be added.

Reply:
I think we should be able to get rid of this entirely in an upcoming refactor of the whole `[validator_]network` system, when we make all the submodules have proper interfaces rather than relay on channels, so I definitely don't want to touch it now.